src/core/qgspallabeling.cpp

```diff
@@ -3416,8 +3407,10 @@ QgsGeometry QgsPalLabeling::prepareGeometry( const QgsGeometry& geometry, QgsRen
     }
   }
 
-  if ( !geom.asGeos() )
-    return QgsGeometry();  // there is something really wrong with the geometry
+  // MD: exporting geometry to GEOS just to see if the geometry is wrong...?
+  // if still needed, we could have a more specific test here
+  //if ( !geom.asGeos() )
```

Comment:
Cool, so I will remove that code - isGeosValid() can handle errors on export to GEOS...